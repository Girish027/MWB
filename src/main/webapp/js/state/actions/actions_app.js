
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
