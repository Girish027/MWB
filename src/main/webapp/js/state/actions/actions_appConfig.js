import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
