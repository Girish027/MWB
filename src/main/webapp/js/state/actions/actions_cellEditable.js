import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
