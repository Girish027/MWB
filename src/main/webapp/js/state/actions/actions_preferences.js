import store from 'state/configureStore';
import getUrl, { pathKey } from 'utils/apiUrls';
import Constants from 'constants/Constants';
import api from 'utils/api';
import * as appActions from './actions_app';
import * as types from './types';

export const technologySelected = ({
  allPreferences = [], clientLevelPreference = {}, isClientAvailable = false, vectorizer, clientType,
}) => ({
  type: types.DEFAULT_TECHNOLOGY,
  allPreferences,
  clientType,
  clientLevelPreference,
  isClientAvailable,
  vectorizer,
});

export const projectPreference = ({
  response,
}) => ({
  type: types.RECEIVE_PROJECT_PREFERENCE,
  response,
});

export const latestVersion = ({
  response,
}) => ({
  type: types.RECEIVE_LATEST_TECHNOLOGY,
  response,
});

export const addOrUpdateTechnology = data => (dispatch, getState) => {
  const state = store.getState();
  const {
    isClientAvailable = false, clientLevelPreference = {}, allPreferences = [], technology,
  } = state.preferences;
  const { projectById } = state.projectListSidebar;
  const projectIdList = Object.keys(projectById);
  const { technologyValue = '', updateExistingModels = false } = data;
  const { id = '' } = clientLevelPreference;
  let isModelsAvailable = false;
  let modelsResponse = allPreferences.filter((record) => record.level === Constants.LEVEL.MODEL);
  let preferenceList = [];
  if (modelsResponse.length > 0) {
    isModelsAvailable = true;
    projectIdList.forEach((project) => {
      const response = modelsResponse.find((record) => record.attribute === project);
      if (!response) {
        preferenceList.push(project);
      }
    });
  } else {
    preferenceList = projectIdList;
  }

  if (isClientAvailable) {
    if (technology === technologyValue) {
      if (updateExistingModels && isModelsAvailable) {
        if (preferenceList.length > 0) {
          dispatch(createModelPreferences({ modelsResponse: preferenceList, technologyValue }));
        }
        dispatch(updateModelTechnology({ technologyValue, modelsResponse }));
      } else if (updateExistingModels) {
        if (preferenceList.length > 0) {
          dispatch(createModelPreferences({ modelsResponse: preferenceList, technologyValue }));
        }
      }
      const technologyType = technology === Constants.MODEL_TECHNOLOGY.USE ? Constants.TENSORFLOW : Constants.N_GRAM;
      const technologyMessage = technologyType === Constants.TENSORFLOW ? Constants.TECHNOLOGY.DIGITAL_TENSORFLOW : Constants.TECHNOLOGY.DIGITAL_NGRAM;
      dispatch(appActions.displayGoodRequestMessage(Constants.SETTINGS_NOTIFICATION_MESSAGE(technologyMessage)));
    } else if (updateExistingModels && isModelsAvailable) {
      dispatch(updateModelTechnology({ technologyValue, modelsResponse }));
      dispatch(updateClientTechnology({ id, technologyValue }));
      if (preferenceList.length > 0) {
        dispatch(createModelPreferences({ modelsResponse: preferenceList, technologyValue }));
      }
    } else {
      if (updateExistingModels && (preferenceList.length > 0)) {
        dispatch(createModelPreferences({ modelsResponse: preferenceList, technologyValue }));
      }
      dispatch(updateClientTechnology({ id, technologyValue }));
    }
  } else if (technologyValue !== '') {
    if (isModelsAvailable) {
      dispatch(updateModelTechnology({ technologyValue, modelsResponse }));
    }
    if ((updateExistingModels) && (preferenceList.length > 0)) {
      dispatch(createModelPreferences({ modelsResponse: preferenceList, technologyValue }));
    }
    dispatch(addClientTechnology({ technologyValue }));
  }
};

export const onUpdateModelTechnologySuccess = (apiResponse, id) => (dispatch, getState) => {
  const state = store.getState();
  const { allPreferences = [], vectorizer = {} } = state.preferences;
  let clientRecord = {};
  let clientType;
  let isClientAvailable = false;
  if (allPreferences && allPreferences.length === 0) {
    allPreferences.push(apiResponse);
  } else {
    let index = allPreferences.findIndex((record) => record.id === id);
    allPreferences.splice(index, 1, apiResponse);
    clientRecord = allPreferences.find((record) => record.level === Constants.LEVEL.CLIENT);
    clientRecord = (clientRecord && (Object.keys(clientRecord).length > 0)) ? clientRecord : {};
  }
  if (clientRecord && (Object.keys(clientRecord).length > 0)) {
    isClientAvailable = true;
    if (Object.keys(vectorizer).length !== 0) {
      const currentClientType = Object.keys(vectorizer).find(key => vectorizer[key] === clientRecord.value);
      clientType = Constants.MODEL_TECHNOLOGY[currentClientType];
    }
  }
  dispatch(technologySelected({
    allPreferences, clientLevelPreference: clientRecord, clientType, isClientAvailable: true, vectorizer,
  }));
};

export const onUpdateModelTechnologyFailure = (error) => (dispatch, getState) => {
  const state = store.getState();
  const { allPreferences = [], vectorizer = {} } = state.preferences;
  let clientLevelPreference = {};
  let isClientAvailable = false;
  let clientType;
  if (allPreferences && allPreferences.length !== 0) {
    const clientRecord = allPreferences.find((record) => record.level === Constants.LEVEL.CLIENT);
    clientLevelPreference = (clientRecord && (Object.keys(clientRecord).length > 0)) ? clientRecord : {};
  }
  if (clientLevelPreference && (Object.keys(clientLevelPreference).length > 0)) {
    isClientAvailable = true;
    if (Object.keys(vectorizer).length !== 0) {
      const currentClientType = Object.keys(vectorizer).find(key => vectorizer[key] === clientLevelPreference.value);
      clientType = Constants.MODEL_TECHNOLOGY[currentClientType];
    }
  }
  dispatch(technologySelected({
    allPreferences, clientLevelPreference, isClientAvailable, vectorizer, clientType,
  }));
};

export const onCreateModelPreferencesSuccess = (apiResponse) => (dispatch, getState) => {
  const state = store.getState();
  const { allPreferences = [], vectorizer = {} } = state.preferences;
  let clientRecord = {};
  let isClientAvailable = false;
  let clientType;
  allPreferences.push(apiResponse);
  clientRecord = allPreferences.find((record) => record.level === Constants.LEVEL.CLIENT);
  clientRecord = (clientRecord && (Object.keys(clientRecord).length > 0)) ? clientRecord : {};
  isClientAvailable = clientRecord.type !== '';
  if (clientRecord && (Object.keys(clientRecord).length > 0)) {
    isClientAvailable = true;
    if (Object.keys(vectorizer).length !== 0) {
      const currentClientType = Object.keys(vectorizer).find(key => vectorizer[key] === clientRecord.value);
      clientType = Constants.MODEL_TECHNOLOGY[currentClientType];
    }
  }
  dispatch(technologySelected({
    allPreferences, clientLevelPreference: clientRecord, clientType, isClientAvailable: true, vectorizer,
  }));
};

export const onCreateModelPreferencesFailure = (error) => (dispatch, getState) => {
  const state = store.getState();
  const { allPreferences = [], vectorizer = {} } = state.preferences;
  let clientLevelPreference = {};
  let isClientAvailable = false;
  let clientType;
  if (allPreferences && allPreferences.length !== 0) {
    const clientRecord = allPreferences.find((record) => record.level === Constants.LEVEL.CLIENT);
    clientLevelPreference = (clientRecord && (Object.keys(clientRecord).length > 0)) ? clientRecord : {};
  }
  if (clientLevelPreference && (Object.keys(clientLevelPreference).length > 0)) {
    isClientAvailable = true;
    if (Object.keys(vectorizer).length !== 0) {
      const currentClientType = Object.keys(vectorizer).find(key => vectorizer[key] === clientLevelPreference.value);
      clientType = Constants.MODEL_TECHNOLOGY[currentClientType];
    }
  }
  dispatch(technologySelected({
    allPreferences, clientLevelPreference, isClientAvailable, vectorizer, clientType,
  }));
};

export const updateModelTechnology = data => (dispatch, getState) => {
  const state = store.getState();
  const { allPreferences = [], vectorizer = {}, clientId } = state.preferences;
  const { modelsResponse, technologyValue = '' } = data;
  const vectorizerId = technologyValue === Constants.MODEL_TECHNOLOGY.USE ? vectorizer.USE : vectorizer.N_GRAM;
  modelsResponse.forEach((record) => {
    const { id } = record;
    let url = getUrl(pathKey.updateTechnology, {
      clientId,
      id,
    });
    let formData = [{
      op: Constants.OP.REPLACE,
      path: '/value',
      value: vectorizerId,
    }];

    api.patch({
      dispatch,
      getState,
      url,
      headers: {
        'Content-Type': 'application/json-patch+json',
      },
      data: formData,
      onApiSuccess: response => onUpdateModelTechnologySuccess(response, id),
      onApiError: error => onUpdateModelTechnologyFailure(error),
    });
  });
};

export const onAddOrUpdateClientTechnologySuccess = (apiResponse = {}) => (dispatch, getState) => {
  const state = store.getState();
  let { allPreferences = [], vectorizer = {} } = state.preferences;
  let clientType;
  if (allPreferences && allPreferences.length === 0) {
    allPreferences.push(apiResponse);
  } else {
    allPreferences = allPreferences.filter((record) => record.level === Constants.LEVEL.MODEL);
    allPreferences.push(apiResponse);
  }
  if (Object.keys(vectorizer).length !== 0) {
    const currentClientType = Object.keys(vectorizer).find(key => vectorizer[key] === apiResponse.value);
    clientType = Constants.MODEL_TECHNOLOGY[currentClientType];
  }
  const type = clientType === Constants.MODEL_TECHNOLOGY.USE ? Constants.TENSORFLOW : Constants.N_GRAM;
  const technologyMessage = type === Constants.TENSORFLOW ? Constants.TECHNOLOGY.DIGITAL_TENSORFLOW : Constants.TECHNOLOGY.DIGITAL_NGRAM;
  dispatch(technologySelected({
    allPreferences, clientLevelPreference: apiResponse, isClientAvailable: true, vectorizer, clientType,
  }));
  dispatch(appActions.displayGoodRequestMessage(Constants.SETTINGS_NOTIFICATION_MESSAGE(technologyMessage)));
};

export const onAddOrUpdateClientTechnologyFailure = (error) => (dispatch, getState) => {
  const state = store.getState();
  const { allPreferences = [], vectorizer = {} } = state.preferences;
  let clientLevelPreference = {};
  let clientType;
  let isClientAvailable = false;
  if (allPreferences && allPreferences.length !== 0) {
    const clientRecord = allPreferences.find((record) => record.level === Constants.LEVEL.CLIENT);
    clientLevelPreference = (clientRecord && (Object.keys(clientRecord).length > 0)) ? clientRecord : {};
  }
  if (clientLevelPreference && (Object.keys(clientLevelPreference).length > 0)) {
    isClientAvailable = true;
    if (Object.keys(vectorizer).length !== 0) {
      const currentClientType = Object.keys(vectorizer).find(key => vectorizer[key] === clientLevelPreference.value);
      clientType = Constants.MODEL_TECHNOLOGY[currentClientType];
    }
  }
  dispatch(technologySelected({
    allPreferences, clientLevelPreference, isClientAvailable, vectorizer, clientType,
  }));
};

export const updateClientTechnology = data => (dispatch, getState) => {
  const state = store.getState();
  const { allPreferences = [], clientId, vectorizer = {} } = state.preferences;
  const { id, technologyValue = '' } = data;
  const vectorizerId = technologyValue === Constants.MODEL_TECHNOLOGY.USE ? vectorizer.USE : vectorizer.N_GRAM;
  if (id === '') {
    dispatch(addClientTechnology({ clientId, technologyValue }));
    return;
  }
  let url = getUrl(pathKey.updateTechnology, {
    clientId,
    id,
  });

  let formData = [{
    op: Constants.OP.REPLACE,
    path: '/value',
    value: vectorizerId,
  }];

  api.patch({
    dispatch,
    getState,
    url,
    headers: {
      'Content-Type': 'application/json-patch+json',
    },
    data: formData,
    onApiSuccess: apiResponse => onAddOrUpdateClientTechnologySuccess(apiResponse),
    onApiError: error => onAddOrUpdateClientTechnologyFailure(error),
  });
};

export const addClientTechnology = data => (dispatch, getState) => {
  const state = store.getState();
  const { allPreferences = [], vectorizer = {}, clientId } = state.preferences;
  const { technologyValue = '' } = data;
  const vectorizerId = technologyValue === Constants.MODEL_TECHNOLOGY.USE ? vectorizer.USE : vectorizer.N_GRAM;
  let url = getUrl(pathKey.addTechnology, {
    clientId,
    setDefault: true,
  });

  const value = {
    level: Constants.LEVEL.CLIENT,
    type: Constants.VECTORIZER_TYPE,
    value: vectorizerId,
    attribute: clientId,
  };

  api.post({
    dispatch,
    getState,
    url,
    data: value,
    onApiSuccess: apiResponse => onAddOrUpdateClientTechnologySuccess(apiResponse),
    onApiError: error => onAddOrUpdateClientTechnologyFailure(error),
  });
};

export const createModelPreferences = data => (dispatch, getState) => {
  const state = store.getState();
  const { allPreferences = [], vectorizer = {}, clientId } = state.preferences;
  const { modelsResponse, technologyValue = '' } = data;
  const vectorizerId = technologyValue === Constants.MODEL_TECHNOLOGY.USE ? vectorizer.USE : vectorizer.N_GRAM;
  modelsResponse.forEach((record) => {
    let url = getUrl(pathKey.addTechnology, {
      clientId,
      setDefault: true,
    });
    const value = {
      level: Constants.LEVEL.MODEL,
      type: Constants.VECTORIZER_TYPE,
      value: vectorizerId,
      attribute: record,
    };

    api.post({
      dispatch,
      getState,
      url,
      data: value,
      onApiSuccess: response => onCreateModelPreferencesSuccess(response),
      onApiError: error => onCreateModelPreferencesFailure(error),
    });
  });
};

export const onGetTechnologySuccess = (data, response = []) => (dispatch, getState) => {
  const state = store.getState();
  if (response.length > 0) {
    const vectorizerResponse = response.filter((record) => record.type === Constants.VECTORIZER_TYPE);
    const clientLevelPreference = vectorizerResponse.find((record) => record.level === Constants.LEVEL.CLIENT);
    const vectorizer = state.preferences.vectorizer;
    let clientType;
    if (clientLevelPreference && (Object.keys(clientLevelPreference).length > 0)) {
      if (Object.keys(vectorizer).length !== 0) {
        const currentClientType = Object.keys(vectorizer).find(key => vectorizer[key] === clientLevelPreference.value);
        clientType = Constants.MODEL_TECHNOLOGY[currentClientType];
      }
      dispatch(technologySelected({
        allPreferences: vectorizerResponse, clientType, clientLevelPreference, isClientAvailable: true, vectorizer,
      }));
    } else {
      dispatch(technologySelected({
        allPreferences: vectorizerResponse, clientType: Constants.MODEL_TECHNOLOGY.N_GRAM, clientLevelPreference: {}, isClientAvailable: false, vectorizer,
      }));
    }
  }
};

export const onGetTechnologyFailure = (error) => (dispatch, getState) => {
  const state = store.getState();
  const vectorizer = state.preferences.vectorizer;
  dispatch(technologySelected({
    clientLevelPreference: {}, isClientAvailable: false, clientType: Constants.MODEL_TECHNOLOGY.N_GRAM, allPreferences: [], vectorizer,
  }));
};

export const getTechnology = data => (dispatch, getState) => {
  const { clientId } = data;
  let url = getUrl(pathKey.getTechnology, {
    clientId,
  });
  api.get({
    dispatch,
    getState,
    url,
    onApiSuccess: response => onGetTechnologySuccess(data, response),
    onApiError: error => onGetTechnologyFailure(error),
  });
};

export const onGetVectorizerSuccess = (response = []) => (dispatch, getState) => {
  const state = getState();
  let { allPreferences = [], clientLevelPreference, isClientAvailable } = state.preferences;
  let clientType;
  if (response.length > 0) {
    const NGramVectorizer = response.find((record) => record.type === Constants.MODEL_TECHNOLOGY.N_GRAM);
    const USEVectorizer = response.find((record) => record.type === Constants.MODEL_TECHNOLOGY.USE && record.isLatest === 1);
    const vectorizer = {
      N_GRAM: NGramVectorizer.id,
      USE: USEVectorizer.id,
    };
    if (clientLevelPreference && (Object.keys(clientLevelPreference).length > 0)) {
      const currentClientType = Object.keys(vectorizer).find(key => vectorizer[key] === clientLevelPreference.value);
      clientType = Constants.MODEL_TECHNOLOGY[currentClientType];
    }
    dispatch(technologySelected({
      vectorizer, allPreferences, clientType, isClientAvailable, clientLevelPreference,
    }));
  }
};

export const onGetVectorizerFailure = (error) => (dispatch, getState) => {
  const state = getState();
  let { allPreferences = [], vectorizer = {} } = state.preferences;
  dispatch(technologySelected({ vectorizer, allPreferences }));
};

export const getVectorizer = data => (dispatch, getState) => {
  let url = getUrl(pathKey.getVectorizer);
  api.get({
    dispatch,
    getState,
    url,
    onApiSuccess: response => onGetVectorizerSuccess(response),
    onApiError: error => onGetVectorizerFailure(error),
  });
};

export const getTechnologyByClientModel = data => (dispatch, getState) => {
  const { clientId, projectId } = data;
  let url = getUrl(pathKey.getVectorizerByClientProject, {
    clientId,
    projectId,
  });
  api.get({
    dispatch,
    getState,
    url,
    onApiSuccess: response => getVectorizerByClientProjectResult(response),
    onApiError: error => getVectorizerByClientProjectResult(),
  });
};

export const getVectorizerByClientProjectResult = (response = {}) => (dispatch, getState) => {
  dispatch(projectPreference({
    response,
  }));
};

export const getVectorizerByTechnology = data => (dispatch, getState) => {
  const { technology } = data;
  let url = getUrl(pathKey.getVectorizerByTechnology, {
    technology,
  });
  api.get({
    dispatch,
    getState,
    url,
    onApiSuccess: response => getVectorizerByTechnologySuccess(response),
    onApiError: error => getVectorizerByTechnologyFailure(error),
  });
};

export const getVectorizerByTechnologySuccess = (response) => (dispatch, getState) => {
  if (response) {
    dispatch(latestVersion({
      response,
    }));
  }
};

export const getVectorizerByTechnologyFailure = (error = {}) => (dispatch, getState) => {
  const response = {};
  dispatch(latestVersion({
    response,
  }));
};
