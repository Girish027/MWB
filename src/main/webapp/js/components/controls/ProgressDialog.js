import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Dialog, DashedSpinner } from '@tfs/ui-components';
import * as actionsApp from 'state/actions/actions_app';
import * as preferenceActions from 'state/actions/actions_preferences';
import { progressDialogLabel } from '../../styles';

export default class ProgressDialog extends Component {
  constructor(props) {
    super(props);
    this.props = props;

    this.onClickClose = this.onClickClose.bind(this);
    this.onClickOk = this.onClickOk.bind(this);
  }

  componentDidMount() {
    const {
      clientId, projectId, dispatch,
    } = this.props;
    if (clientId && projectId) {
      dispatch(preferenceActions.getTechnologyByClientModel({ clientId, projectId }));
    }
  }

  onClickClose() {
    const { dispatch } = this.props;
    dispatch(actionsApp.modalDialogChange(null));
  }

  onClickOk() {
    const { history, onOk } = this.props;
    onOk(history);
  }

  render() {
    const {
      message, closeIconVisible, header, showFooter, showHeader,
      okVisible, cancelVisible, okChildren, cancelChildren,
      styleOverride, showSpinner,
    } = this.props;

    return (
      <div>
        <Dialog
          size="small"
          headerChildren={header}
          isOpen
          okVisible={okVisible}
          cancelVisible={cancelVisible}
          closeIconVisible={closeIconVisible}
          onClickClose={this.onClickClose}
          styleOverride={styleOverride}
          okChildren={okChildren}
          cancelChildren={cancelChildren}
          onClickOk={this.onClickOk}
          onClickCancel={this.onClickClose}
          showHeader={showHeader}
          showFooter={showFooter}
        >
          <div>
            <p style={progressDialogLabel}>
              {message}
            </p>
            {showSpinner && <DashedSpinner height="50" width="50" fill="#004C97" />}
          </div>
        </Dialog>
      </div>
    );
  }
}

ProgressDialog.propTypes = {
  message: PropTypes.string,
  closeIconVisible: PropTypes.bool,
  showHeader: PropTypes.bool,
  showFooter: PropTypes.bool,
  okVisible: PropTypes.bool,
  cancelVisible: PropTypes.bool,
  showSpinner: PropTypes.bool,
  header: PropTypes.string,
  okChildren: PropTypes.string,
  cancelChildren: PropTypes.string,
  styleOverride: PropTypes.object,
  onOK: PropTypes.func,
};

ProgressDialog.defaultProps = {
  message: '',
  closeIconVisible: false,
  showHeader: false,
  showFooter: false,
  okVisible: false,
  cancelVisible: false,
  showSpinner: true,
  string: '',
  okChildren: '',
  cancelChildren: '',
  styleOverride: {
    childContainer: {
      margin: 'auto',
    },
  },
  onOk: () => {},
};
