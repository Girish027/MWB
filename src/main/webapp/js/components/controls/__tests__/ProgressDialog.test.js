import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as actionsApp from 'state/actions/actions_app';
import ProgressDialog from 'components/controls/ProgressDialog';
import * as actionsPreferences from 'state/actions/actions_preferences';

describe('<ProgressDialog />', () => {
  let wrapper;

  describe('Creating an instance with no props', () => {
    beforeEach(() => {
      wrapper = shallow(<ProgressDialog />);
    });
    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    test('Snapshots should render correctly', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Creating an instance with test props', () => {
    const props = {
      message: 'Loading',
      clientId: '123',
      projectId: '123',
      closeIconVisible: true,
      dispatch: () => {},
    };

    beforeAll(() => {
      props.dispatch = jest.fn();
      actionsApp.modalDialogChange = jest.fn(() => 'called modalDialogChange');
      actionsPreferences.getTechnologyByClientModel = jest.fn(() => 'getTechnologyByClientModel');
    });

    beforeEach(() => {
      wrapper = shallow(<ProgressDialog
        {...props}
      />);
    });

    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    test('Snapshots should render correctly', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('should dispatch an action when closeIcon is clicked', () => {
      wrapper.instance().onClickClose();
      expect(actionsApp.modalDialogChange).toHaveBeenCalled;
    });

    test('should dispatch action to save vectorizer', () => {
      wrapper.instance().componentDidMount();
      expect(actionsPreferences.getTechnologyByClientModel).toHaveBeenCalled();
      expect(props.dispatch).toHaveBeenCalledWith('getTechnologyByClientModel');
    });
  });
});
