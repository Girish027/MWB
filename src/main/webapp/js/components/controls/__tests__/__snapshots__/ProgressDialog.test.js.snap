// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressDialog /> Creating an instance with no props Snapshots should render correctly 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren=""
    cancelVisible={false}
    closeIconVisible={false}
    isOpen={true}
    okChildren=""
    okVisible={false}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showFooter={false}
    showHeader={false}
    size="small"
    styleOverride={
      Object {
        "childContainer": Object {
          "margin": "auto",
        },
      }
    }
  >
    <div>
      <p
        style={
          Object {
            "color": "#313F54",
            "marginBottom": "25px",
            "textAlign": "left",
          }
        }
      />
      <Memo(withTheme(DashedSpinner))
        fill="#004C97"
        height="50"
        width="50"
      />
    </div>
  </withTheme(DialogComponent)>
</div>
`;

exports[`<ProgressDialog /> Creating an instance with test props Snapshots should render correctly 1`] = `
<div>
  <withTheme(DialogComponent)
    cancelChildren=""
    cancelVisible={false}
    closeIconVisible={true}
    isOpen={true}
    okChildren=""
    okVisible={false}
    onClickCancel={[Function]}
    onClickClose={[Function]}
    onClickOk={[Function]}
    showFooter={false}
    showHeader={false}
    size="small"
    styleOverride={
      Object {
        "childContainer": Object {
          "margin": "auto",
        },
      }
    }
  >
    <div>
      <p
        style={
          Object {
            "color": "#313F54",
            "marginBottom": "25px",
            "textAlign": "left",
          }
        }
      >
        Loading
      </p>
      <Memo(withTheme(DashedSpinner))
        fill="#004C97"
        height="50"
        width="50"
      />
    </div>
  </withTheme(DialogComponent)>
</div>
`;
